File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

bool cmp (string a, string b) {
    return a + b > b + a;
}

string solution(vector<int> numbers) {
    string answer = "";
    vector<string> tmp;
    for (auto num : numbers) {
        tmp.push_back(to_string(num));
    }
    
    sort(tmp.begin(), tmp.end(), cmp);
    
    for (auto t : tmp) {
        answer += t;
    }
    if (answer[0] == '0')
        return "0";
    return answer;
}

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.java
import java.util.List;
import java.util.ArrayList;

class Solution {
    public String solution(int[] numbers) {
        String answer = "";
        List<String> list = new ArrayList<>();
        
        for (int n : numbers) {
            list.add(Integer.toString(n));
        }
        list.sort((a,b) -> (b+a).compareTo(a+b));
        
        if ("0".equals(list.get(0)))
            return "0";
        for (String s : list)
            answer += s;
        
        return answer;
    }
}